eng/testing/tests.mobile.targets

```diff
@@ -186,13 +213,15 @@
       <_runnerFilesToPublish Include="$(AppleTestRunnerDir)*" Condition="'$(TargetOS)' == 'iOS' or '$(TargetOS)' == 'tvOS'" />
       <_runnerFilesToPublish Include="$(WasmTestRunnerDir)*" Condition="'$(TargetOS)' == 'Browser'" />
 
-      <!-- Exclude xunit assemblies as those should be resolved by our own package references -->
-      <ResolvedFileToPublish Include="@(_runnerFilesToPublish)" Condition="!$([System.String]::Copy('%(Filename)').StartsWith('xunit.'))" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />
+      <ResolvedFileToPublish Include="@(_runnerFilesToPublish)" Condition="'$(EnableAggressiveTrimming)' == 'true'" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />
+      <ResolvedFileToPublish Include="@(_runnerFilesToPublish)" Condition="'$(EnableAggressiveTrimming)' != 'true' and !$([System.String]::Copy('%(Filename)').StartsWith('xunit.'))" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />
     </ItemGroup>
```

Comment:
Why is EnableAggressiveTrimming condition needed?

Reply:
This was added to preserve the original functionality (when EnableAggressiveTrimming is false) where only non-xunit files were resolved to avoid a Double Write https://github.com/dotnet/runtime/pull/37938/files?file-filters%5B%5D=.targets#r441043014. 

I'm not sure what scenario resulted in double writes, but for testing `System.Buffers` on Browser wasm Debug ```<ResolvedFileToPublish Include="@(_runnerFilesToPublish)" RelativePath="%(FileName)%(Extension)" CopyToPublishDirectory="PreserveNewest" PostprocessAssembly="true" />```
I didn't see any double writes with respect to `xunit.` files. Either the double write has been fixed elsewhere or testing a specific test suite for Browser wasm Debug doesn't repro the double write?

Reply:
Reverted the change that conditioned `ResolvedFileToPublish` on `EnableAggressiveTrimming`.